# GSM model for E. coli

In [1]:
import numpy as np
from cobra.io import read_sbml_model

In [2]:
model = read_sbml_model('iML1515.xml.gz')

We want to look at the model. See if it has fluxes

In [3]:
solution = model.optimize()
solution.fluxes

CYTDK2     0.00000
XPPT       0.00000
HXPRT      0.00000
NDPK5      0.00000
SHK3Dr     0.33424
            ...   
MPTS       0.00000
MOCOS      0.00000
BMOGDS2    0.00000
FESD2s     0.00000
OCTNLL     0.00000
Name: fluxes, Length: 2712, dtype: float64

In [4]:
model.summary()

We want to find the metabolite Purtescine in the model

In [5]:
for metabolite in model.metabolites.query('Putrescine', 'name'):
    print(metabolite.name)
    print(metabolite.id)

Putrescine
ptrc_p
Putrescine
ptrc_c
Putrescine
ptrc_e


In [6]:
model.metabolites.get_by_id('ptrc_c') # Putrescine

<Metabolite ptrc_c at 0x7fd0fe1e1da0>

The reaction that produces purtescine from ornithine is native to E. coli

In [7]:
model.reactions.ORNDC

<Reaction ORNDC at 0x7fd0fce067f0>

We want to check the flux

In [8]:
with model:
    model.objective = model.reactions.ORNDC
    print(model.optimize().objective_value)  # We produce Purtescine

41.09128205128201


In [9]:
with model:
    model.objective = model.reactions.EX_ptrc_e
    print(model.optimize().objective_value)  # We produce Purtescine.EX_ptrc_e

9.56477064220184


# Adding enzyme 2.6.1.13

In [10]:
from cobra import Reaction, Metabolite

To make E. Coli produce putrescine we need to add the reaction of the enzyme 2.6.1.13.
This is the same as we added in C. Glutamicum.

In [11]:
new_reaction_1 = Reaction('OAT')  # OAT -> (o)rnithine (a)mino(t)ransferase

In [12]:
for metabolite in model.metabolites.query('5-semialdehyde', 'name'):
    print(metabolite.name)
    print(metabolite.id)

N-Acetyl-L-glutamate 5-semialdehyde
acg5sa_c
N2-Succinyl-L-glutamate 5-semialdehyde
sucgsa_c
L-Glutamate 5-semialdehyde
glu5sa_c


In [13]:
model.metabolites.get_by_id('glu5sa_c') # L-glutamate 5-semialdehyde

<Metabolite glu5sa_c at 0x7fd0fe37be80>

We see that the two reactions L-glutamate 5-semialdehyde is part of correspond to the ones seen in the KEGG pathway map. We now need to find the other metabolites that play a role in conversion of L-glutamate 5-semialdehyde to ornithine.

In [14]:
for metabolite in model.metabolites.query('Ornithine', 'name'):
    print(metabolite.name)
    print(metabolite.id)

Ornithine
orn_e
Ornithine
orn_p
Ornithine
orn_c


In [15]:
model.metabolites.get_by_id('orn_c') # Ornithine

<Metabolite orn_c at 0x7fd0fe296828>

In [16]:
for metabolite in model.metabolites.query('Oxoglutarate', 'name'):
    print(metabolite.name)
    print(metabolite.id)

2-Oxoglutarate
akg_e
2-Oxoglutarate
akg_c
2-Oxoglutarate
akg_p


In [17]:
model.metabolites.get_by_id('akg_c') # 2-Oxoglutarate

<Metabolite akg_c at 0x7fd0fe5dc518>

In [18]:
for metabolite in model.metabolites.query('L-Glutamate', 'name'):
    print(metabolite.name)
    print(metabolite.id)

L-Glutamate 1-semialdehyde
glu1sa_c
L-Glutamate
glu__L_c
L-Glutamate 5-semialdehyde
glu5sa_c
L-Glutamate
glu__L_e
L-Glutamate
glu__L_p
L-Glutamate 5-phosphate
glu5p_c


In [19]:
model.metabolites.get_by_id('glu__L_c') # L-Glutamate

<Metabolite glu__L_c at 0x7fd0fe377e10>

Now we add the reaction to produce ornithine

In [20]:
new_reaction_1.add_metabolites({model.metabolites.get_by_id('glu5sa_c'): -1, # L-glutamate 5-semialdehyde
                              model.metabolites.get_by_id('glu__L_c'): -1, # L-glutamate 
                              model.metabolites.get_by_id('orn_c'): 1, # L-ornithine
                              model.metabolites.get_by_id('akg_c'): 1 # 2-oxoglutarate
                             })

In [21]:
print(new_reaction_1.build_reaction_string())

glu5sa_c + glu__L_c --> akg_c + orn_c


In [22]:
model.add_reactions([new_reaction_1])
model.reactions.OAT

<Reaction OAT at 0x7fd0fd089048>

In [23]:
with model:
    model.objective = model.reactions.OAT
    print(model.optimize().objective_value)  # We produce L-ornithine

53.02333333333331


We want to see production rate of putrescine now after we have added a new path via glutamate

In [24]:
with model:
    model.objective = model.reactions.ORNDC
    print(model.optimize().objective_value)  # We produce Purtescine

45.787428571428464


In [25]:
with model:
    model.objective = model.reactions.EX_ptrc_e
    print(model.optimize().objective_value)  # We produce Purtescine

9.92914285714285


We see that the production rate of putrescine increases from 41.09 to 45.79

# Adding enzyme 3.5.3.1

Uses L-arginine and H2O to make L-ornithine and Urea

In [26]:
for metabolite in model.metabolites.query('L-Arginine', 'name'):
    print(metabolite.name)
    print(metabolite.id)

L-Arginine
arg__L_c
L-Arginine
arg__L_e
L-Arginine
arg__L_p


In [27]:
for metabolite in model.metabolites.query('Urea', 'name'):
    print(metabolite.name)
    print(metabolite.id)

Urea CH4N2O
urea_c
Urea CH4N2O
urea_p
Urea CH4N2O
urea_e


In [28]:
for metabolite in model.metabolites.query('H2O', 'name'):
    print(metabolite.name)
    print(metabolite.id)

H2O H2O
h2o_p
H2O H2O
h2o_c
H2O H2O
h2o_e


In [29]:
new_reaction_2 = Reaction('AAH')  # AAH -> L-arginine amidinohydrolase

In [30]:
new_reaction_2.add_metabolites({model.metabolites.get_by_id('arg__L_c'): -1, # L-arginine
                              model.metabolites.get_by_id('h2o_c'): -1, # H2O 
                              model.metabolites.get_by_id('orn_c'): 1, # L-ornithine
                              model.metabolites.get_by_id('urea_c'): 1 # Urea
                             })

In [31]:
print(new_reaction_2.build_reaction_string())

arg__L_c + h2o_c --> orn_c + urea_c


In [32]:
model.add_reactions([new_reaction_2])
model.reactions.AAH

<Reaction AAH at 0x7fd0fd066b38>

In [33]:
with model:
    model.objective = model.reactions.ORNDC
    print(model.optimize().objective_value)  # We produce Purtescine

45.787428571428464


In [34]:
with model:
    model.objective = model.reactions.EX_ptrc_e
    print(model.optimize().objective_value)

9.929142857142857
